frontend/src/store/modules/sqlEditor.ts

```diff
@@ -303,89 +274,45 @@ const actions = {
       state.queryHistoryList.filter((t: QueryHistory) => t.id !== id)
     );
   },
-  async createSavedQuery(
-    { commit, state }: any,
-    { name, statement }: { name: string; statement: string }
-  ): Promise<SavedQuery> {
-    const data = (
-      await axios.post(`/api/savedquery`, {
-        data: {
-          type: "createSavedQuery",
-          attributes: {
-            name,
-            statement,
-          },
-        },
-      })
-    ).data;
-    const newSavedQuery = convertSavedQuery(data.data, data.included);
+  /**
+   * Set the connection by tab info
+   * @param param
+   * @param payload
+   */
+  setCurrentConnectionByTab(
```

Comment:
We should avoid method like this as it relies on the internal state like rootGetters["tab/currentTab"]

A better way is to pass in the tab (which is also more appropriate to the method name).

Reply:
I just want to reuse the function, Extract setConnectionByTab as a composable function